### PR TITLE
use xpu-smi to dump used memory

### PR DESCRIPTION
**Pull request by sywangyi**
xpu use "ZE_AFFINITY_MASK" to control card, usage is like CUDA_VISIBLE_DEVICES

# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by sywangyi** [comment]
@yao-matrix

**Comment by sywangyi** [comment]
@Narsil please help review

File: server/text_generation_server/utils/import_utils.py [hunk 1]
```diff
@@ -19,8 +20,12 @@ def get_cuda_free_memory(device, memory_fraction):
 
 
 def get_xpu_free_memory(device, memory_fraction):
-    total_gpu_memory = torch.xpu.get_device_properties(device).total_memory
-    free_memory = int(total_gpu_memory * 0.5)
+    total_memory = torch.xpu.get_device_properties(device).total_memory
+    device_id = str(device)[4]
```
[the remark this answers is not in the document]
Comment (sywangyi) [reply server/text_generation_server/utils/import_utils.py]: we are pushing, suppose will be ready EOY 2024

[the remark this answers is not in the document]
Comment (sywangyi) [reply server/text_generation_server/utils/import_utils.py]: currently, we use xpu-smi to get as a temp solution

[the remark this answers is not in the document]
Comment (danieldk) [reply server/text_generation_server/utils/import_utils.py]: Awesome, thanks!

**Review by danieldk — APPROVED** [review]
Looks good to me, but I'll wait for @Narsil or someone else to review as well, since I'm not really familiar with the XPU support.

**Review by Narsil — APPROVED** [review]
LGTM

**Pull request merged**